### PR TITLE
Fixed some variables/tasks to allow upgrade correctly

### LINKED ISSUE #30: [BUG] ClickHouse server fails to start after upgrading
### Description

After Clickhouse version being updated to version 22.7.3.5 with this role, a bug was found when the role try to restart clickhouse-server.service.

### Steps to Reproduce
1. Change deb repository and key:

  - clickhouse_deb_repo: "deb https://packages.clickhouse.com/deb {{  clickhouse_release_type }} main"

  - clickhouse_deb_keyid: 8919F6BD2B48D754
2. Change clickhouse version:

- clickhouse_version: 22.7.3.5

4. Run this role in a playbook

**Expected behavior:**
The ClickHouse server restarted successfully.

**Actual behavior:**
The ClickHouse server failed to restart:
```
# service clickhouse-server status
● clickhouse-server.service - ClickHouse Server (analytic DBMS for big data)
     Loaded: loaded (/etc/systemd/system/clickhouse-server.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Wed 2022-08-24 06:31:21 MSK; 1s ago
    Process: 17550 ExecStart=/usr/bin/clickhouse-server --config=/etc/clickhouse-server/config.xml --pid-file=/run/clickhouse-server/clickhouse-server.pid (code=exited, status=203/EXEC)
   Main PID: 17550 (code=exited, status=203/EXEC)
        CPU: 10ms
```

**Reproduces how often:** 
Clickhouse upgrade has been tested with different versions with the same result

### Additional Information
Bug resolved manually following this thread https://github.com/ClickHouse/ClickHouse/issues/37099#issuecomment-1125315887

```
rm /etc/systemd/system/clickhouse-server.service
systemctl daemon-reload
service clickhouse-server restart
```

### PR DESCRIPTION
**Pull request by lpeman**
### Description of the Change

- Update deb repository and key to upgrade to new versions
- Update clickhouse-server.service template 

### Benefits
Upgrade clickhouse version 
### Applicable Issues

Resolves https://github.com/idealista/clickhouse_role/issues/30

### CODE REVIEW

**Review by amanzanotejon — APPROVED** [review]

**Review by jmonterrubio — APPROVED** [review]

**Review by Plozano94 — APPROVED** [review]

File: templates/clickhouse-server.service.j2 [hunk 1]
```diff
@@ -1,29 +0,0 @@
-{{ ansible_managed | comment }}
```
Comment (ultraheroe) [inline templates/clickhouse-server.service.j2]: For maintaining the same behavior till now, I would prefer to make the necessary changes than delete and use the default (with no option to customize) service file.

It's seems that we only had to update some lines from the service file like CapabilityBoundingSet and maybe the After and Wants lines like in here https://github.com/ClickHouse/ClickHouse/blob/b4b603e0b57bdd1ce373ca278b21a7267ffff2ee/packages/clickhouse-server.service

File: tasks/service-clickhouse.yml [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
 
-- name: CLICKHOUSE | Copy systemd unit configuration file
-  template:
-    src: "{{ clickhouse_service_template_path }}"
-    dest: "/etc/systemd/system/{{ clickhouse_service }}.service"
-    mode: 0644
-    owner: root
-    group: root
-  notify: reload-clickhouse
+- name: CLICKHOUSE | Remove file (delete file if exists)
```
Comment (ultraheroe) [inline tasks/service-clickhouse.yml]: If we update the service template with the new settings this change isn't necessary

**Review by ultraheroe — CHANGES_REQUESTED** [review]
Don't forget to update the changelog! 😉

File: defaults/main.yml [hunk 3]
```diff
@@ -47,17 +47,16 @@ clickhouse_prerequisite_repos:
...
-clickhouse_deb_keyid: E0C56BD4
+clickhouse_deb_keyid: 8919F6BD2B48D754
 clickhouse_packages:
   - procps
   - "clickhouse-client={{clickhouse_version}}"
   - "clickhouse-common-static={{clickhouse_version}}"
   - "clickhouse-server={{clickhouse_version}}"
 
 # Config templates
-clickhouse_service_template_path: clickhouse-server.service.j2
```
Comment (ultraheroe) [inline defaults/main.yml]: I think this var may stay for service config, as is used in the _CLICKHOUSE | Copy systemd unit configuration file_ task

File: tasks/service-clickhouse.yml (hunk 2 above)
Comment (ultraheroe) [reply tasks/service-clickhouse.yml]: solved by @lpeman

File: templates/clickhouse-server.service.j2 (hunk 1 above)
Comment (ultraheroe) [reply templates/clickhouse-server.service.j2]: solved by @lpeman

**Review by ultraheroe — APPROVED** [review]

**Pull request merged**